### PR TITLE
Preserve `.git` suffix in repository names when generating SourceLink URLs

### LINKED ISSUE #1618: Azure Repos: repository names ending with .git are stripped when generating SourceLink URLs
When RepositoryUrl points to an Azure Repos repository whose actual name ends with `.git`,
GetSourceLinkUrl strips the `.git` suffix from the generated SourceLink URL.

Example:

Input:
https://dev.azure.com/org/project/_git/repo.git

Actual:
https://dev.azure.com/org/project/_apis/git/repositories/repo/items?api-version=1.0&versionType=commit&version=<sha>&path=/*

Expected:
https://dev.azure.com/org/project/_apis/git/repositories/repo.git/items?api-version=1.0&versionType=commit&version=<sha>&path=/*

This causes 404 responses for repositories whose actual name includes `.git`.

I have a fix and regression tests prepared.

### PR DESCRIPTION
**Pull request by Eales**
Fixes #1618

Fix an issue where repository names ending with `.git` are incorrectly
normalized when generating SourceLink URLs for Azure Repos.

When the repository name itself ends with `.git`, the shared Git task may
normalize the repository path and strip the suffix before it reaches the
Azure Repos provider. Azure DevOps REST APIs require the exact repository
name when constructing URLs such as:

https://dev.azure.com/org/project/_apis/git/repositories/{repositoryName}/items

Example:

Repository URL:
https://dev.azure.com/org/project/_git/repo.git

Generated SourceLink URL before:
.../_apis/git/repositories/repo/items?...

Expected:
.../_apis/git/repositories/repo.git/items?...

This change fixes the issue in the Azure Repos provider by resolving the
repository name directly from the original repository URI instead of relying
on the normalized relative path produced by the shared Git task.

Changes:
- resolve repository name from `gitUri.GetPath()` in the Azure Repos provider
- keep existing behavior of the shared Git SourceLink task unchanged
- add regression tests for repositories whose names end with `.git`

### CODE REVIEW

File: src/SourceLink.AzureRepos.Git.UnitTests/GetSourceLinkUrlTests.cs [hunk 1]
```diff
@@ -285,5 +285,55 @@ public void VisualStudioHost_ImplicitHost_DeafultCollection_Project_Team(string
...
+        {
+            var urlWithoutDotGit = ExecuteDevAzureCom("https://dev.azure.com/org/project/_git/repo");
+            var urlWithDotGit = ExecuteDevAzureCom("https://dev.azure.com/org/project/_git/repo.git");
+
+            Assert.True(
+                !string.Equals(urlWithoutDotGit, urlWithDotGit, StringComparison.Ordinal),
+                $"Repository name is ignored: URLs are identical.\n" +
+                $"Input without .git: https://dev.azure.com/org/project/_git/repo\n" +
+                $"Input with .git:    https://dev.azure.com/org/project/_git/repo.git\n" +
+                $"Output:             {urlWithoutDotGit}");
```
[the remark this answers is not in the document]
Comment (Eales) [reply src/SourceLink.AzureRepos.Git.UnitTests/GetSourceLinkUrlTests.cs:299]: Keeping the explicit assertion message here since this is a regression test.
The additional context makes failures easier to diagnose than a plain Assert.NotEqual.

**Comment by Eales** [comment]
@dotnet-policy-service agree

**Comment by Eales** [comment]
This fixes Source Link URL generation for Azure DevOps repositories whose actual repository name ends with `.git`.

The current logic trims the `.git` suffix too early, which produces incorrect URLs such as:
`.../_apis/git/repositories/soneta/items`
instead of:
`.../_apis/git/repositories/soneta.git/items`.

Since Source Link is included in the .NET SDK starting with .NET 8, this change may also be relevant for the following SDK branches.

**Comment by Eales** [comment]
Friendly ping 👋

@tmat could you please take a look when you have a moment?
This fixes #1618 and includes regression tests for Azure Repos repository names ending with `.git`.

Happy to adjust anything if needed.

**Comment by Eales** [comment]
Friendly follow-up 👋

@tmat just wanted to check if you had a chance to look at this PR.

No rush of course — just want to make sure it didn’t slip through the cracks. Happy to update anything if needed 🙂

**Comment by Eales** [comment]
Hi 👋

Adding a couple of folks for visibility 🙂

@tmat @EgorBo @jkoritzinsky

This PR fixes #1618 and adds regression tests for Azure Repos `.git` edge case.

It should be low-risk and fully covered by tests.  
Would appreciate a quick review or guidance on whether this is good to merge.

Happy to update anything if needed 🙏

File: src/SourceLink.AzureRepos.Git.UnitTests/GetSourceLinkUrlTests.cs [hunk 2]
```diff
@@ -72,7 +72,6 @@ public void BadUrl(string domainAndAccount, string host)
 
             var result = task.Execute();
 
-            // ERROR : The value of SourceRoot.RepositoryUrl with identity '/src/' is invalid: 'http://account.visualstudio.com/_git'""
```
Comment (tmat) [inline src/SourceLink.AzureRepos.Git.UnitTests/GetSourceLinkUrlTests.cs:75]: Why removing these comments?

File: src/SourceLink.AzureRepos.Git/GetSourceLinkUrl.cs [hunk 3]
```diff
@@ -32,7 +32,7 @@ protected override Uri GetDefaultContentUriFromRepositoryUri(Uri repositoryUri)
 
         protected override string? BuildSourceLinkUrl(Uri contentUri, Uri gitUri, string relativeUrl, string revisionId, ITaskItem? hostItem)
         {
-            if (!AzureDevOpsUrlParser.TryParseHostedHttp(gitUri.GetHost(), relativeUrl, out var projectPath, out var repositoryName))
+            if (!AzureDevOpsUrlParser.TryParseHostedHttp(gitUri.GetHost(), gitUri.GetPath(), out var projectPath, out var repositoryName))
```
Comment (tmat) [inline src/SourceLink.AzureRepos.Git/GetSourceLinkUrl.cs:38]: It'd be useful to add a comment that unlike other git providers AzDO does not allow optional `.git` suffix and thus we are not using relativeUrl here.

Copilot research:

```
Provider behavior

┌──────────────────┬──────────────────────────┬────────────────────────────────────┐
│ Provider         │ .git suffix supported?   │ Required?                          │
├──────────────────┼──────────────────────────┼────────────────────────────────────┤
│ GitHub           │ ✅                       │ No (optional)                      │
├──────────────────┼──────────────────────────┼────────────────────────────────────┤
│ GitLab           │ ✅                       │ No (optional)                      │
├──────────────────┼──────────────────────────┼────────────────────────────────────┤
│ Bitbucket        │ ✅                       │ No (optional)                      │
├──────────────────┼──────────────────────────┼────────────────────────────────────┤
│ Gitea            │ ✅                       │ No (optional)                      │
├──────────────────┼──────────────────────────┼────────────────────────────────────┤
│ Gitee            │ ✅                       │ No (optional)                      │
├──────────────────┼──────────────────────────┼────────────────────────────────────┤
│ Azure DevOps     │ ❌                       │ N/A — adding .git can cause 404s   │
└──────────────────┴──────────────────────────┴────────────────────────────────────┘

Why Azure DevOps is different

Azure DevOps uses a different URL routing scheme (/org/project/_git/repo) tied to its project-based API. It doesn't strip or remap the 
.git suffix. This isn't a spec violation — it's a provider-specific design choice. Every other major provider happens to support it as
a convention, but none are required to.
```

**Comment by Eales** [comment]
I’ve addressed the feedback in this PR and pushed the requested changes.

Could you please take another look?

@tmat

**Review by tmat — APPROVED** [review]
